content/en/blog/_posts/2021-12-15-pod-security-admission-psa-beta.md

```diff
@@ -0,0 +1,620 @@
+--->
+
+<!-- TODO: is this a BUG?:
+It errors with runtimeClassNames: [] and runs without it, tested on both alphav1/betav1 apis.
+
+2021-11-16T02:59:09.121870759Z stderr F E1116 02:59:09.121285       1 run.go:120] "command failed" err="failed to initialize admission: couldn't init admission plugin \"PodSecurity\": strict decoding error: unknown field \"runtimeClassNames\""
+ --->
+
+
+```shell
+cat <<EOF > pod-security.yaml
+apiVersion: apiserver.config.k8s.io/v1
+kind: AdmissionConfiguration
+plugins:
+- name: PodSecurity
+  configuration:
+    apiVersion: pod-security.admission.config.k8s.io/v1beta1
+    kind: PodSecurityConfiguration
+    defaults:
+      enforce: "baseline"
+      enforce-version: "latest"
+      audit: "baseline"
+      audit-version: "latest"
+      warn: "restricted"
+      warn-version: "latest"
+    exemptions:
+      # Array of authenticated usernames to exempt.
+      usernames: []
+      # Array of runtime class names to exempt.
+      runtimeClasses: []
+      # Array of namespaces to exempt.
+      namespaces: [kube-system]
+EOF
+```
+
+We now have a default baseline policy, next pass it to the kind configuration to enable the `--admission-control-config-file` API server argument and pass the policy file.
+
+```shell
+cat <<EOF > kind-config.yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+        # enable admission-control-config flag on the API server
+        extraArgs:
+          admission-control-config-file: /etc/kubernetes/policies/pod-security.yaml
+        # mount new file / directories on the control plane
+        extraVolumes:
+          - name: policies
+            hostPath: /etc/kubernetes/policies
+            mountPath: /etc/kubernetes/policies
+            readOnly: true
+            pathType: "DirectoryOrCreate"
+  # mount the local file on the control plane
+  extraMounts:
+  - hostPath: ./pod-security.yaml
+    containerPath: /etc/kubernetes/policies/pod-security.yaml
+    readOnly: true
+EOF
+```
+
+```shell
+kind create cluster --image kindest/node:v1.23.0 --config kind-config.yaml
+```
+
+Let's look at the default namespace.
+
+```shell
+kubectl describe namespace default
+```
+
+The output is similar to this:
+
+```shell
+Name:         default
+Labels:       kubernetes.io/metadata.name=default
+Annotations:  <none>
+Status:       Active
+
+No resource quota.
+
+No LimitRange resource.
+```
+
+Let's create a new namespace and see if the labels apply there.
+
+```shell
+kubectl create namespace test-defaults
+kubectl describe namespace test-defaults
+```
+
+Same.
+
+```shell
+Name:         test-defaults
+Labels:       kubernetes.io/metadata.name=test-defaults
+Annotations:  <none>
+Status:       Active
+
+No resource quota.
+
+No LimitRange resource.
+```
+
+Can we deploy a workload that is privileged?
```

Comment:
Avoid writing “we” if the sense you're using it in includes the reader; that form can be confusing (and we avoid it even more in the docs).